src/js/tech/html5.js

```diff
@@ -86,6 +87,12 @@ class Html5 extends Tech {
    * @method dispose
    */
   dispose() {
+    let tt = this.el().textTracks;
+
+    tt.removeEventListener('change', Fn.bind(this, this.handleTextTrackChange));
```

Comment:
I believe the solution in this case really is a setTimeout around the removal of tracks. Either that or create a remove handler that gets added on `addtrack` and bound to a particular track which then gets removed when that particular track gets removed. But I think that solution might be too clever.